panedrlite/panedrlite/panedr.py

```diff
@@ -428,11 +466,30 @@ def read_edr(path, verbose=False):
               end='', file=sys.stderr)
         print('\n{} frame read in {:.2f} seconds'.format(ifr, end - begin),
               file=sys.stderr)
-
     return all_energies, all_names, times
 
 
+
 def edr_to_df(path: str, verbose: bool = False):
```

Comment:
```suggestion
def edr_to_df(path: str, verbose: bool = False) -> pd.DataFrame:
```

Reply:
This is a bit complicated because pandas is an optional dependency. If pandas is not installed, the module can't be imported if the type hint is present. I could put the function definition into a try-except statement, but then I would lose the custom ImportError message. What's the best thing to do here?

Reply:
that being said, one of these solutions might work: https://stackoverflow.com/questions/61384752/how-to-type-hint-with-an-optional-import

Reply:
I took one of the suggestions from there and it seems to work :)

Reply:
Spoke to soon - I didn't fully think this through, the solution I tried yesterday won't work I'm afraid. I think we'll have to hold back on annotating the return type of `edr_to_df` for now, but the function name and doc string are pretty self-explanatory, and mypy wasn't working with that anyway yet 

mypy error message: "error: Skipping analyzing "pandas": module is installed, but missing library stubs or py.typed marker
panedrlite/panedrlite/panedr.py:59: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports"

Reply:
I would say don't worry about it. We lived without typed code for Python 0-3.whatever so I'm sure well be fine without it.  Sorry for leading you down the garden path.